wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -868,13 +870,25 @@ data Account = Account
     , accWalletId  :: !WalletId
     } deriving (Show, Ord, Eq, Generic)
 
+-- | Datatype wrapping addresses for per-field endpoint
+newtype AccountAddresses = AccountAddresses
+    { acaAddresses :: [WalletAddress]
+    } deriving (Show, Ord, Eq, Generic)
+
+-- | Datatype wrapping balance for per-field endpoint
+newtype AccountBalance = AccountBalance
+    { acbAmount    :: V1 Core.Coin
+    } deriving (Show, Ord, Eq, Generic)
+
```

Comment:
This is really open to debate. This will preserve a similar structure than the one used for account. For instance, retrieving the account balance will return a JSON with:

```json
{
    "status": "success",
    "data": {
        "amount": 14
    },
    "metadata": ...
}
```

Another option would be to remove one level of nesting and put the value directly under `data`:

```json
{
    "status": "success",
    "data": 14,
    "metadata": ...
}
```

I am really open to discuss this.

Reply:
What is the argument for the flattened one? I have no idea how this would affect usage in various client libraries, maybe there is some specific argument for either one. I wouldn't instinctively argue against the nested approach though.

Reply:
Could one enforce the "keeping things roughly the same" part? When first seeing this PR I came to think about lenses and GraphQL…

Reply:
I always thought about `data` as a container of, well, data, so it would feel unnatural to me to go from `data :: Object` to `data :: Value`, if you see what you mean. I suspect I lean towards symmetry and consistency, and as such having `data` always index a JSON object seems like the best approach, to me, but of course, that's just my opinion 😉

Reply:
I agree that consistency and convention are important (especially without types), and right now, as far as JS is concerned, we've got a type like `{ data :: Object }`. Deviating from that would a potential source of runtime errors.